### PR TITLE
Restructuring React

### PR DESCRIPTION
**Pull request by imcodingideas**
I need to fix tests..

### CODE REVIEW

**Comment by user512** [comment]
Test back to green but one test is skipped, don't merge yet, I might add a few more test.

**Comment by imcodingideas** [comment]
@user512 did you see the other PR?

**Comment by user512** [comment]
Yes, @imcodingideas, I am looking into why his build failed, by just looking at his code, it should not be failing.

**Review by user512 — COMMENTED** [review]
@imcodingideas skipped test is fixed, do you mind letting me know why we do rename ClassTable to Home?

`ClassTable` seems to be a good enough name where it tells us the component responsibility.

File: client/src/components/main/App.js [hunk 1]
```diff
@@ -46,11 +47,20 @@ export default class App extends Component {
           studentLength={this.state.students.length}
           fetchStudentsFromParent={this.fetchStudentList}
         />
-        <ClassTable
-          handleDelete={this.handleDelete}
-          students={this.state.students}
-          errors={this.state.errors}
-        />
+        <MainContent>
```
Comment (user512) [inline client/src/components/main/App.js:50]: I think `AddStudentFrom` probably should lives in `MainContent`, maybe even inside `ClassTable`, but maybe we can move this once we introduce a global state so we no longer have to pass props thru multiple layer of components.

**Pull request merged**